pennylane/_device.py

```diff
@@ -62,12 +62,20 @@ def __init__(self, wires=1, shots=1000):
 
     def __repr__(self):
         """String representation."""
-        return "{}.\nInstance: ".format(self.__module__, self.__class__.__name__, self.name)
+        return "<{} ({}, wires={}, shots={}) at {}>".format(
+            self.__class__.__name__, self.short_name, self.num_wires, self.shots, hex(id(self))
```

Comment:
```suggestion
        return "<{} device (wires={}, shots={}) at {}>".format(
            self.__class__.__name__, self.num_wires, self.shots, hex(id(self))
```

Reply:
Would you prefer to see the short name or the class name, if we had to pick one? e.g.,

```
<BasicAerDevice(wires=4, shots=1024) at 0x7f05108a37f0>
```

or 

```
<qiskit.basicaer(wires=4, shots=1024) at 0x7f05108a37f0>
```

Reply:
The class name looks more 'right' to me